utils/frame/sub-du/Cargo.toml

```diff
@@ -0,0 +1,26 @@
+[package]
+name = "sub-du"
+version = "0.1.0"
+authors = ["kianenigma <kian.peymani@gmail.com>"]
+edition = "2021"
+
+[dependencies]
+codec = { package = "parity-scale-codec", version = "3.0.0" }
+async-std = { version = "1.12.0", features = ["attributes"] }
+jsonrpsee = { version = "0.15.1", features = ["jsonrpsee-types", "jsonrpsee-http-client", "jsonrpsee-ws-client"] }
+separator = "0.4.1"
+ansi_term = "0.12.1"
+env_logger = "0.9"
+log = "0.4.17"
+structopt = "0.3"
+
+sub-storage = { path = "../sub-storage", features = ["helpers"] }
```

Comment:
Next step: this has to go away entirely. We should directly use the JSONRPSEE client and helpers. @niklasad1 can help here if needed.

Reply:
It's all good, will take a stab at it.

Reply:
I don't know if we really need `http` but the clients implements two traits `ClientT and SubscriptionClientT` that you can abstract of such as `impl ClientT` 

However, these traits are not object-safe so you can't use it as a `trait object`.

Maybe https://github.com/paritytech/substrate/pull/12212 will come in handy here ^^

Reply:
@niklasad1 shall I introduce a get_metadata call there as well?

Reply:
I reckon that you might need something similar as https://github.com/paritytech/polkadot/blob/master/utils/staking-miner/src/rpc.rs#L163-#L178